plugins/processors/aws/ec2/ec2.go

```diff
@@ -29,9 +29,22 @@ type AwsEc2Processor struct {
 	ImdsTags         []string        `toml:"imds_tags"`
 	EC2Tags          []string        `toml:"ec2_tags"`
 	Timeout          config.Duration `toml:"timeout"`
+	CacheTTL         config.Duration `toml:"cache_ttl"`
 	Ordered          bool            `toml:"ordered"`
 	MaxParallelCalls int             `toml:"max_parallel_calls"`
 	Log              telegraf.Logger `toml:"-"`
+	TagCacheSize     int             `toml:"tag_cache_size"`
+	LogCacheStats    bool            `toml:"log_cache_stats"`
+
+	// We use a single cache because telegraf's AddTag function models this.
+	//
+	// A user can specify a list of both EC2 tags and IMDS tags. The items in this list can, technically, be the same.
+	// This will result in a situation where the EC2 Tag's value will override the IMDS tags value.
+	//
+	// Though this is undesirable, it is unavoidable because the AddTag function does not support this case.
+	// Users should be aware that you should avoid using IMDS tags as EC2 tags because the EC2 tags will always "win"
+	// due to them being processed in this plugin *after* IMDS tags.
```

Comment:
Can you please add this statement to the `README.md` file as users typically do not check the code. :-)